src/main/java/com/adyen/model/checkout/details/ApplePayDetails.java

```diff
@@ -0,0 +1,198 @@
+/*
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Java API Library
+ *
+ * Copyright (c) 2020 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ */
+
+package com.adyen.model.checkout.details;
+
+import java.util.Objects;
+
+import com.adyen.model.checkout.PaymentMethodDetails;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
+
+/**
+ * ApplePayDetails
+ */
+
+public class ApplePayDetails implements PaymentMethodDetails {
+    /**
+     * Possible types
+     */
+    public static final String APPLE_PAY = "applepay";
```

Comment:
constant name APPLEPAY is more appropriate

Reply:
Apple Pay payment method is two words, while Dotpay is a single one

Reply:
changed to APPLEPAY for consistency with txvariant